compass/landice/tests/mismipplus/setup_mesh.py

```diff
@@ -0,0 +1,301 @@
+    y = y / 1.e3      # m to km
+    B0 = -150.        # m
+    B2 = -728.8       # m
+    B4 = 343.91       # m
+    B6 = -50.57       # m
+    x_bar = 300.      # km
+    x_tilde = x / x_bar
+    dc = 500.         # m
+    fc = 4.           # km
+    wc = 24.          # km
+    Ly = 80.          # km
+    Bmax = -720.      # m
+
+    B_x = B0 + B2 * x_tilde**2 + B4 * x_tilde**4 + B6 * x_tilde**6
+    B_y = dc / (1 + np.exp(-2 * (y - Ly / 2 - wc) / fc)) +\
+        dc / (1 + np.exp(2 * (y - Ly / 2 + wc) / fc))
+
+    Bsum = B_x + B_y
+
+    z_b = np.maximum(Bsum, Bmax)  # Eqn 1. (Asay-Davis et al. 2016)
+
+    return z_b
+
+
+def _setup_MISMPPlus_IC(config, logger, filename):
+    """
+    Add the inital condition for the MISMIP+ spinup to the given MPAS mesh file
+
+    Parameters
+    ----------
+    config : comass.config.CompassConfigParser
+        Configuration options for this test case
+
+    logger : logging.Logger
+        A logger for output from the step
+
+    filename : str
+        NetCDF file to place the MISMIP+ ICs in
+
+    """
+
+    # Hard code some parameters from Table 1. of Asay-Davis et al. 2016
+    accum = 0.3       # m^{-1}
+    C = 3.160e6       # Pa m^{-1/3} s^{1/3}
+    rhoi = 918        # kg m^{-3}
+    spy = 31556926    # s a^{-1}
+    xcalve = 640.e3   # m
+
+    # Read parameters from the .cfg file
+    section = config['mesh']
+    nVertLevels = section.getint('levels')
+    init_thickness = section.getfloat('init_thickness')
+
+    # open the file
+    src = xr.open_dataset(filename)
+
+    # Use `.loc[:]` for setting the initial conditions  since we are setting
+    # the fields with scalar values. This ensures values are properly broadcast
+    # against the fields coordinates
+
+    # Set the bedTopography
+    src['bedTopography'].loc[:] = _mismipplus_bed(src.xCell, src.yCell)
+
+    # Set the ice thickness
+    src['thickness'].loc[:] = xr.where(src.xCell < xcalve, init_thickness, 0.)
+
+    # Convert SMB from m/yr to kg/m2/s
+    accum *= rhoi / spy
+    # Set the surface mass balance
+    src['sfcMassBal'].loc[:] = xr.where(src.xCell > xcalve, accum, -100.)
+
+    # Boolean masks for indices which correspond to the N/S boundary of mesh
+    mask = (src.yCell == src.yCell.min()) | (src.yCell == src.yCell.max())
+    # Set the velocity boundary conditions
+    # NOTE: `.variable` is needed so that coordinates are properly broadcast
+    #        due to a bug in xarray, which was resolved as of 2023.9.0
+    src['dirichletVelocityMask'].loc[:] = xr.where(mask, 1, 0).variable
+
+    # Skipping WHL step of setting the initial velocities to zero
+    # since they are already zero.
+
+    # convert to MPAS units
+    C /= spy**(1.0 / 3.0)
+
+    # Set the effectivePressure using a Weertman power law
+    src['effectivePressure'].loc[:] = C
+
+    # Set the layerThicknessFractions so that layers are evenly distributed
+    src['layerThicknessFractions'].loc[:] = 1.0 / float(nVertLevels)
```

Comment:
I don't think there is any reason why this needs to be case (I don't see a specification in the MISMIP+ paper), and if not I would prefer to stick with our default nonuniform vertical spacing.  I'm guessing this was specified in the old test case before we had implemented a standard nonuniform spacing option in our landice mesh creation script.  @trhille, what do you think? (if you interpolate from our archived MISMIP+ 2km state, this will be a difference to be aware of)

Reply:
I agree that we don't need to have uniform layer thickness. When interpolating from the archived state, I don't know if we will be interpolating 3D velocity, since the point of this (at least for now) is to use the velocity solver. We could also punt on this for now and have a config setting for 'uniform' or 'glimmer' layer spacing.

Reply:
Great, I'll add a config option. How does `vetical_layer_distribution` with the options `uniform` or `glimmer` sound? I'll leave the default as `glimmer` so the `SetupMesh` step doesn't alter the vertical layer distribution, as created by the `create_landice_grid_from_generic_MPAS_grid.py`, by default.